Raven.Database/FileSystem/Storage/Esent/Schema/Updates/From08To09.cs

```diff
@@ -0,0 +1,189 @@
+        {
+        }
+
+        public void Update(Session session, JET_DBID dbid, Action<string> output)
+        {
+            var pagesTableName = "pages";
+
+            var newTableName = pagesTableName + "_new";
+
+            try
+            {
+                Api.JetDeleteTable(session, dbid, newTableName);
+            }
+            catch (Exception)
+            {
+                //if there is no such table - then it is not important
+                //this is a precaution against partially failed upgrade process
+            }
+
+            SchemaCreator.CreatePagesTable(dbid, newTableName, session);
+
+            var buffer = new byte[StorageConstants.MaxPageSize];
+            var bookMarkBuffer = new byte[SystemParameters.BookmarkMost];
+
+            using (var usage = new Table(session, dbid, "usage", OpenTableGrbit.None))
+            {
+                // let's create index in usages table necessary for migration purposes
+                // we need to seek by page_id in order to update value in 'usages' table after
+                // writing a page to new table where it will get new id
+
+                var indexDef = "+page_id\0\0";
+                try
+                {
+                    Api.JetDeleteIndex(session, usage, "by_page_id");
+                }
+                catch (Exception)
+                {
+                    //if there is no such index - then it is not important
+                    //this is a precaution against partially failed upgrade process
+                }
+
+                Api.JetCreateIndex2(session, usage, new[]
+                {
+                    new JET_INDEXCREATE
+                    {
+                        szIndexName = "by_page_id",
+                        cbKey = indexDef.Length,
+                        cbKeyMost = SystemParameters.KeyMost,
+                        cbVarSegMac = SystemParameters.KeyMost,
+                        szKey = indexDef,
+                        grbit = CreateIndexGrbit.None,
+                        ulDensity = 80
+                    }
+                }, 1);
+
+                // commit index creation
+
+                Api.JetCommitTransaction(session, CommitTransactionGrbit.LazyFlush);
+                Api.JetBeginTransaction2(session, BeginTransactionGrbit.None);
+
+                var usageColumns = Api.GetColumnDictionary(session, usage);
+
+                using (var src = new Table(session, dbid, pagesTableName, OpenTableGrbit.None))
+                using (var dst = new Table(session, dbid, newTableName, OpenTableGrbit.None))
+                { 
+                    Api.MoveBeforeFirst(session, src);
+                    Api.MoveBeforeFirst(session, dst);
+
+                    var srcColumns = Api.GetColumnDictionary(session, src);
+                    var dstColumns = Api.GetColumnDictionary(session, dst);
+
+                    int rows = 0;
+
+                    while (Api.TryMoveNext(session, src))
```

Comment:
This is very likely going to take a LONG time, no?
Can we try doing this in place? with add / remove column?

Reply:
Not an option. It's primary key column - it cannot be removed

Reply:
Okay :-(